actors/builtin/miner/miner_state.go

```diff
@@ -118,24 +118,31 @@ type WorkerKeyChange struct {
 	EffectiveAt abi.ChainEpoch
 }
 
+// Information provided by a miner when pre-committing a sector.
 type SectorPreCommitInfo struct {
 	SealProof     abi.RegisteredSealProof
 	SectorNumber  abi.SectorNumber
 	SealedCID     cid.Cid // CommR
 	SealRandEpoch abi.ChainEpoch
 	DealIDs       []abi.DealID
-	Expiration    abi.ChainEpoch // Sector Expiration
+	Expiration    abi.ChainEpoch
 }
 
+// Information stored on-chain for a pre-committed sector.
 type SectorPreCommitOnChainInfo struct {
 	Info             SectorPreCommitInfo
 	PreCommitDeposit abi.TokenAmount
 	PreCommitEpoch   abi.ChainEpoch
 }
 
+// Information stored on-chain for a proven sector.
 type SectorOnChainInfo struct {
-	Info               SectorPreCommitInfo
-	ActivationEpoch    abi.ChainEpoch // Epoch at which SectorProveCommit is accepted
+	SectorNumber       abi.SectorNumber
+	SealProof          abi.RegisteredSealProof // The seal proof type implies the PoSt proof/s
+	SealedCID          cid.Cid                 // CommR
+	DealIDs            []abi.DealID
+	Activation         abi.ChainEpoch // Epoch during which the sector proof was accepted
+	Expiration         abi.ChainEpoch // Epoch during which the sector expires
```

Comment:
nit: You may want to consider putting these fields into an embedded struct so we can share a common datastructure (easier to add common fields). But I'm not sure if it's worth it in this case.

Reply:
Thanks. I'm not going to in this case because (a) it cost a (hundred-mega) byte when encoded, and (b) I fear will encourage unwanted re-used of that structure which led to this kind of thing in the first place.

Reply:
Hm. I guess our CBOR library doesn't act like `encoding/json` (`encoding/json` will inline embedded structs).